Assets/LeapMotion/Core/Scripts/Hands/RiggedFinger.cs

```diff
@@ -12,40 +12,90 @@
-     * Only set to true when mesh is not visible
-     */
+    /// <summary>
+    /// Allows the mesh to be stretched to align with finger joint positions.
+    /// Only set to true when mesh is not visible.
+    /// </summary>
     [HideInInspector]
     public bool deformPosition = false;
 
+    [HideInInspector]
+    public bool scaleLastFingerBone = false;
+
     public Vector3 modelFingerPointing = Vector3.forward;
     public Vector3 modelPalmFacing = -Vector3.up;
 
     public Quaternion Reorientation() {
       return Quaternion.Inverse(Quaternion.LookRotation(modelFingerPointing, -modelPalmFacing));
     }
 
-    private RiggedHand riggedHand;
 
-    /** Updates the bone rotations. */
+    /// <summary>
+    /// Fingertip lengths for the standard edit-time hand.
+    /// </summary>
+    private static float[] s_standardFingertipLengths = null;
+    static RiggedFinger() {
+      // Calculate standard fingertip lengths.
+      s_standardFingertipLengths = new float[5];
+      var testHand = TestHandFactory.MakeTestHand(isLeft: true,
+                           unitType: TestHandFactory.UnitType.UnityUnits);
+      for (int i = 0; i < 5; i++) {
+        var fingertipBone = testHand.Fingers[i].bones[3];
+        s_standardFingertipLengths[i] = fingertipBone.Length;
+      }
+    }
+
+    /// <summary>
+    /// Updates model bone positions and rotations based on tracked hand data.
+    /// </summary>
     public override void UpdateFinger() {
       for (int i = 0; i < bones.Length; ++i) {
         if (bones[i] != null) {
           bones[i].rotation = GetBoneRotation(i) * Reorientation();
           if (deformPosition) {
-            bones[i].position = GetJointPosition(i);
+            var boneRootPos = GetJointPosition(i);
+            bones[i].position = boneRootPos;
+
+            if (i == 3 && scaleLastFingerBone) {
+              // Set fingertip base bone scale to match the bone length to the fingertip.
+              // This will only scale correctly if the model was constructed to match
+              // the standard "test" edit-time hand model from the TestHandFactory.
+              var boneTipPos = GetJointPosition(i + 1);
+              var boneVec = boneTipPos - boneRootPos;
+              var boneLen = boneVec.magnitude;
+              var standardLen = s_standardFingertipLengths[(int)this.fingerType];
+              var newScale = bones[i].transform.localScale;
+              var lengthComponentIdx = getLargestComponentIndex(modelFingerPointing);
+              newScale[lengthComponentIdx] = boneLen / standardLen;
+              bones[i].transform.localScale = newScale;
+            }
           }
         }
       }
     }
+
+    private int getLargestComponentIndex(Vector3 pointingVector) {
+      var largestValue = 0f;
+      var largestIdx = 0;
+      for (int i = 0; i < 3; i++) {
+        var testValue = pointingVector[i];
+        if (Mathf.Abs(testValue) > largestValue) {
+          largestIdx = i;
```

Comment:
Er I just realized this isn't a proper 'get largest component index' since `largestValue` is never assigned.  This is actually get the last component that is non-zero.  Which is _usually_ correct but if there are subtle deviations in the pointing vector can make the answer wrong.  

(Like a vector of 0.999, 0, 0.001)

Reply:
Woops! Nice catch. pushing fix

Reply:
pushed